test/units/modules/cloud/amazon/placebo_fixtures.py

```diff
@@ -0,0 +1,192 @@
+>         asg_module.create_autoscaling_group(connection, module)
+>     .... asserts based on module state/exceptions ....
+
+In more advanced cases, use unrecorded resource fixtures to fill in ARNs/IDs of
+things modules depend on, such as:
+
+> def test_create_in_vpc(placeboify, scratch_vpc):
+>     connection = placeboify.client('autoscaling')
+>     module = FakeModule(name='test-asg-created',
+>         min_size=0, max_size=0, desired_capacity=0,
+>         availability_zones=[s['az'] for s in scratch_vpc['subnets']],
+>         vpc_zone_identifier=[s['id'] for s in scratch_vpc['subnets']],
+>     )
+>     ..... so on and so forth ....
+"""
+
+
+@pytest.fixture
+def placeboify(request, monkeypatch):
+    """This fixture puts a recording/replaying harness around `boto3_conn`
+
+    Placeboify patches the `boto3_conn` function in ec2 module_utils to return
+    a boto3 session that in recording or replaying mode, depending on the
+    PLACEBO_RECORD environment variable. Unset PLACEBO_RECORD (the common case
+    for just running tests) will put placebo in replay mode, set PLACEBO_RECORD
+    to any value to turn off replay & operate on real AWS resources.
+
+    The recorded sessions are stored in the test file's directory, under the
+    namespace `placebo_recordings/{testfile name}/{test function name}` to
+    distinguish them.
+    """
+    session = boto3.Session()
+
+    recordings_path = os.path.join(
+        request.fspath.dirname,
+        'placebo_recordings',
+        request.fspath.basename.replace('.py', ''),
+        request.function.func_name
+        # remove the test_ prefix from the function & file name
+    ).replace('test_', '')
+
+    try:
+        # make sure the directory for placebo test recordings is available
+        os.makedirs(recordings_path)
+    except OSError as e:
+        if e.errno != os.errno.EEXIST:
+            raise
+
+    pill = placebo.attach(session, data_path=recordings_path)
+    if os.getenv('PLACEBO_RECORD'):
+        pill.record()
+    else:
+        pill.playback()
+
+    def boto3_middleman_connection(module, conn_type, resource, region, **kwargs):
+        if conn_type != 'client':
+            #TODO support resource-based connections
+            raise ValueError('Mocker only supports client, not %s' % conn_type)
+        return session.client(resource, region_name=region)
+
+    import ansible.module_utils.ec2
+    monkeypatch.setattr(
+        ansible.module_utils.ec2,
+        'boto3_conn',
+        boto3_middleman_connection,
+    )
+    yield session
+
+    # tear down
+    pill.stop()
+
+
+@pytest.fixture(scope='module')
+def basic_launch_config():
+    """Create an EC2 launch config whose creation *is not* recorded and return its name
+
+    This fixture is module-scoped, since launch configs are immutable and this
+    can be reused for many tests.
+    """
+    if not os.getenv('PLACEBO_RECORD'):
+        yield 'pytest_basic_lc'
+        return
```

Comment:
No, because if the empty return isn't there, then this function would continue on and try to connect to AWS even if recording is turned intentionally off. It's there to exit early if placebo is in replay mode.